### PR TITLE
Port neo-gui

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

**Comment by erikzhang** [comment]
Ready to review.

**Comment by shargon** [comment]
The GUI Forms, are just a copy of `neo-gui`?

**Comment by erikzhang** [comment]
Yes. But I also removed some obsolete features. No new feature has been added.

**Comment by erikzhang** [comment]
neo-gui repository will be removed.

File: neo-gui/neo-gui.csproj [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    <TargetFramework>netcoreapp3.0</TargetFramework>
+    <RootNamespace>Neo</RootNamespace>
+    <UseWindowsForms>true</UseWindowsForms>
+    <Company>The Neo Project</Company>
+    <Product>Neo.GUI</Product>
+    <Description>Neo.GUI</Description>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\neo-cli\neo-cli.csproj" />
```
[the remark this answers is not in the document]
Comment (erikzhang) [reply neo-gui/neo-gui.csproj]: My plan is to reuse the shared function between cli and gui. So I will keep the reference.

**Comment by erikzhang** [comment]
The latest neo master branch is not compatible to the old testnet. (And will never be compatible to it until the new testnet launched.) If you want to test it (and all other PRs), you must test it in a private net built by yourself.

File: neo-gui/neo-gui.csproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (shargon) [reply neo-gui/neo-gui.csproj]: Maybe we should move this shared methods to `neo`

[the remark this answers is not in the document]
Comment (erikzhang) [reply neo-gui/neo-gui.csproj]: Some of them can be moved. Another reason is to allow users to use command line in GUI.

**Comment by erikzhang** [comment]
Is anyone reviewing it?

**Comment by shargon** [comment]
I am, give me one day more please

**Comment by lock9** [comment]
Hi @erikzhang , I reviewed it already, but it is not possible to properly test because it is not possible to deploy _any_ network. We need to fix this ASAP.
According to @belane, the root cause was this PR: https://github.com/neo-project/neo/pull/1249.
I already asked people for a current working configuration, but nobody has it and those who tried informed me that the current code is broken.

**Comment by erikzhang** [comment]
@lock9 We need #491 first.

**Comment by lock9** [comment]
Great, I will check that. Thanks

File: neo-gui/GUI/MainForm.cs [hunk 2]
```diff
@@ -0,0 +1,578 @@
...
+            if (Service.CurrentWallet is null) return;
+            if (!check_nep5_balance || persistence_span < TimeSpan.FromSeconds(2)) return;
+            UInt160[] addresses = Service.CurrentWallet.GetAccounts().Select(p => p.ScriptHash).ToArray();
+            using SnapshotView snapshot = Blockchain.Singleton.GetSnapshot();
+            foreach (UInt160 assetId in NEP5Watched)
+            {
+                byte[] script;
+                using (ScriptBuilder sb = new ScriptBuilder())
+                {
+                    for (int i = addresses.Length - 1; i >= 0; i++)
```
Comment (shargon) [inline neo-gui/GUI/MainForm.cs]:
```suggestion
                    for (int i = addresses.Length - 1; i >= 0; i--)
```

File: neo-gui/GUI/MainForm.cs [hunk 3]
```diff
@@ -0,0 +1,578 @@
...
+            dialog.ShowDialog();
+        }
+
+        private void optionsToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+        }
+
+        private void 官网WToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            Process.Start("https://neo.org/");
```
Comment (shargon) [inline neo-gui/GUI/MainForm.cs]: I can't open the browser in my Windows
image

I found this issue https://github.com/dotnet/corefx/issues/10361

```suggestion
            OpenBrowser("https://neo.org/");
        }

        public static void OpenBrowser(string url)
        {
            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
            {
                Process.Start(new ProcessStartInfo("cmd", $"/c start {url}"));
            }
            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
            {
                Process.Start("xdg-open", url);
            }
            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
            {
                Process.Start("open", url);
            }
        }
        
```

**Comment by vncoelho** [comment]
I just saw the PR now, @erikzhang.
I had a filter for neo-gui, since never used it.

However, this look a good innitiave and nice name for the project. We support neo-node and you did a good choice.

**Pull request merged**